frontend/src/pages/RequestorPage/RequestPanel/BottomToolbar/CopyAsCurl.tsx

```diff
@@ -0,0 +1,96 @@
+import { CheckIcon, CopyIcon } from "@radix-ui/react-icons";
+import { useEffect, useRef, useState } from "react";
+
+import { Button } from "@/components/ui/button";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { RequestorState } from "../../reducer";
+import { getBodyValue, getContentTypeHeader } from "./utils";
+
+export type CopyAsCurlProps = Pick<
+  RequestorState,
+  "body" | "method" | "path" | "requestHeaders" | "queryParams"
+>;
+
+/**
+ * Copy the current request as a cURL command to the clipboard.
+ */
+export function CopyAsCurl({
+  body,
+  method,
+  path,
+  queryParams,
+  requestHeaders,
+}: CopyAsCurlProps) {
+  const [isCopied, setIsCopied] = useState(false);
+  const timeout = useRef<ReturnType<typeof setTimeout>>();
+  const isUnsupportedBodyType =
+    body.type === "file" || (body.type === "form-data" && body.isMultipart);
+
+  const handleCopy = async () => {
+    const payload = getBodyValue({ body, method });
+
+    // As we don't support automatic body type switching yet, we now manually
+    // set the initial header the correspoinding content type based on the
+    // method & body. Otherwise, it will be empty.
+    const initialHeader = payload ? getContentTypeHeader(body) : "";
```

Comment:
Depends on
- #168

Reply:
Putting PR back into draft as above's PR is merged, resulting in duplicate headers being added to the cURL command.